independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanResolver.java

```diff
@@ -111,7 +111,14 @@ boolean matchesNoBoxing(Type requiredType, Type beanType) {
                     throw new IllegalArgumentException("Invalid argument combination " + requiredType + "; " + beanType);
                 }
                 for (int i = 0; i < requiredTypeArguments.size(); i++) {
-                    if (!parametersMatch(requiredTypeArguments.get(i), beanTypeArguments.get(i))) {
+                    Type requiredTypeArg = requiredTypeArguments.get(i);
+                    Type beanTypeArg = beanTypeArguments.get(i);
+                    if (isActualType(requiredTypeArg) && isActualType(beanTypeArg)
+                            && beanTypeArg.name().equals(DotNames.OBJECT)) {
+                        // required type = FooTyped<Long> and bean type is FooTyped<Object>, that is a match
```

Comment:
We don't support `FooType<Number>` though?

Reply:
That should already work I think (should better re-check this).
The problem in this PR was mainly raw types which arc automagically converted to from `FooTyped` to `FooTyped<Object>`

Reply:
Wrong...that actually shouldn't work and it doesn't. Neither on master nor on this PR.
Nonetheless, this PR, I'll put that in a separate comment.